Assets/PC2D/Scripts/PlatformerMotor2D.cs

```diff
@@ -3321,5 +3275,155 @@ void OnDrawGizmosSelected()
         }
     }
 
+    private void ChangeState(MotorState newState) {
+        // no change...
+        if (motorState == newState)
+        {
+            return;
+        }
+
+        if (IsSlipping())
+        {
+            if (onSlippingEnd != null)
+            {
+                onSlippingEnd();
+            }
+        }
+        if (IsJumping())
+        {
+            if (onJumpEnd != null)
+            {
+                onJumpEnd();
+            }
+        }
+        if (IsDashing())
+        {
+            if (onDashEnd != null)
+            {
+                onDashEnd();
+            }
+        }
+
+        // set
+        motorState = newState;
+
+        if (IsSlipping())
+        {
+            if (onSlipping != null)
+            {
+                onSlipping();
+            }
+        }
+        if (IsJumping())
+        {
+            if (onJump != null)
+            {
+                onJump();
+            }
+        }
+        if (IsDashing())
+        {
+            if (onDash != null)
+            {
+                onDash();
+            }
+        }
+    }
+
+    //
+    // Quering motor
+    //
+    #region Public

```

Comment:
There's already a public section up above. Place them there and document them with /// since they will be used outside the motor.